corehq/apps/userreports/models.py

```diff
@@ -605,13 +605,16 @@ def is_static(self):
 STATIC_PREFIX = 'static-'
 CUSTOM_REPORT_PREFIX = 'custom-'
 
-StaticDataSourceMetadata = namedtuple('StaticDataSourceMetadata', 'id path domain')
-StaticReportMetadata = namedtuple('StaticReportMetadata', 'id path domain')
-
 
 class StaticDataSourceConfiguration(JsonObject):
     """
-    For custom data sources maintained in the repository
+    For custom data sources maintained in the repository.
+
+    This class keeps the full list of static data source configurations relevant to the
+    current environment in memory and upon requests builds a new data source configuration
```

Comment:
I don't think I went through all static data sources and reports when I first implemented the relevant environment property. May be worth doing that at some point? I think all the recent ones should have that variable

Reply:
https://github.com/dimagi/commcare-hq/pull/21163